scripts/quests/outlands/Like_Shining_Leggings.lua

```diff
@@ -0,0 +1,108 @@
+---------------------------------------------
+-- Like Shining Leggings
+---------------------------------------------
+-- Log ID: 5 (Outlands), Quest ID: 139
+-- Heizo : !pos -1 -5 25 252
+---------------------------------------------
+require('scripts/globals/zone')
+require('scripts/globals/quests')
+require("scripts/globals/titles")
+require('scripts/globals/npc_util')
+require('scripts/globals/interaction/quest')
+---------------------------------------------
+local ID = require("scripts/zones/Norg/IDs")
+---------------------------------------------
+local quest = Quest:new(xi.quest.log_id.OUTLANDS, xi.quest.id.outlands.LIKE_SHINING_LEGGINGS)
+local leggingsVarName = "shiningLeggings_nb";
+local requiredLeggings = 10
+
+quest.reward =
+{
+    fameArea = xi.quest.fame_area.NORG,
+    fame     = 100,
+    title    = xi.title.LOOKS_GOOD_IN_LEGGINGS,
+    item     = xi.items.SCROLL_OF_DOKUMORI_ICHI,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return (status == QUEST_AVAILABLE and player:getFameLevel(xi.quest.fame_area.NORG) >= 3)
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] = quest:progressEvent(127),
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    quest:begin(player) -- Start Like Shining Leggings
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORG] =
+        {
+            ['Heizo'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(128, player:getCharVar(leggingsVarName))-- Update the player on the number of Rusty Leggings they have turned in
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local curNumTradedIn = player:getCharVar(leggingsVarName)
+                    local numTradingInNow = trade:getItemQty(xi.items.RUSTY_LEGGINGS);
+                    local remainderNeeded = utils.clamp(requiredLeggings - curNumTradedIn, 0, requiredLeggings);
+
+                    local clampedTrade = math.min(numTradingInNow, remainderNeeded) -- Only take the amount needed to complete the quest and nothing more
+
+                    if npcUtil.tradeHas(trade, { { xi.items.RUSTY_LEGGINGS, clampedTrade } }) then
+                        local totalWithTrade = curNumTradedIn + clampedTrade
+
+                        if totalWithTrade >= 10 then
+                            return quest:progressEvent(129) -- Complete the quest
+                        elseif totalWithTrade <= 9 then
+                            player:setCharVar(leggingsVarName, totalWithTrade) -- Keep track of traded so far
+                            player:confirmTrade()
```

Comment:
I thought about it, but my fear was if anyone was already in the middle of this quest and I changed the tracking var then they could lose progress.

I guess I could copy any existing progress from the old into the new if the var already exists on a player. Thoughts?

Reply:
Ah yes, that is a great thought process. The special deployment considerations would be to place account for that. In this instance you could do something like 
`UPDATE char_vars SET varname = 'Quest[5][139]Count' WHERE varname = "shiningLeggings_nb";` 
`UPDATE char_vars SET varname = 'Quest[5][138]Count' WHERE varname = "shiningSubligar_nb";` 
(or whatever it was previously from the NPC lua's. )

Reply:
also remember that Quest[5][139]Prog 0 also counts as a usable var, for future reference. Instead of incrementing to 1 out the gate, (sometimes not needed) you can `quest:getVar(player, 'Prog') == 0  `

Reply:
Ah that's makes sense, definitely good to know. I'll make the change and amend my special deployment considerations section.